pages/signin/index.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { useCallback, useMemo, useState } from "react";
+import Head from "next/head";
+import {
+  Box,
+  Container,
+  TextField,
+  Typography,
+  Button,
+} from "@material-ui/core";
+import { gql, useMutation } from "@apollo/client";
+import { useRouter } from "next/router";
+
+const SEND_SIGN_TOKEN = gql`
+  mutation signin($email: String!, $username: String!) {
+    signin(email: $email, username: $username) {
+      message
+      token
+    }
+  }
+`;
+
+const IndexPage = () => {
+  const router = useRouter();
+  const [userEmail, setMail] = useState("");
+  const [userName, setName] = useState("");
+  const [errorMail, isErrorMail] = useState(false);
+  const [errorName, isErrorName] = useState(false);
+  const [messageErrorMail, setMessageErrorMail]= useState("");
+  const [messageErrorName, setMessageErrorName]= useState("");
+
+  const ableToSignIn = useMemo(() => {
+    return (
+      userEmail &&
+      userName &&
+      /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/.test(
+        userEmail
+      )
+    );
+  }, [userEmail, userName]);
+
+  const [addUser] = useMutation(SEND_SIGN_TOKEN);
+
+  const onClick = useCallback(() => {
+    if (!userEmail || !userName) return;
```

Comment:
We do not need this validation because we disable the button when is not able to sign in